File: Lab1/Lab1_p2.py
import RPi.GPIO as GPIO
import time

LEDPin1 = 16
LEDPin2 = 18
LEDPin3 = 22

def setup():
        GPIO.setmode(GPIO.BOARD)
        GPIO.setup(LEDPin1, GPIO.OUT)
        GPIO.setup(LEDPin2, GPIO.OUT)
        GPIO.setup(LEDPin3, GPIO.OUT)
        GPIO.output(LEDPin1, GPIO.LOW)
        GPIO.output(LEDPin2, GPIO.LOW)
        GPIO.output(LEDPin3, GPIO.LOW)

def loop():
        # light led 1
        print 'LED 1 on'
        GPIO.output(LEDPin1, GPIO.HIGH)
        time.sleep(0.5)
        print 'LED 1 off'
        GPIO.output(LEDPin1, GPIO.LOW)
        # light led 2
        print 'LED 2 on'
        GPIO.output(LEDPin2, GPIO.HIGH)
        time.sleep(0.5)
        print 'LED 2 off'
        GPIO.output(LEDPin2, GPIO.LOW)
        # light led 3
        print 'LED 3 on'
        GPIO.output(LEDPin3, GPIO.HIGH)
        time.sleep(0.5)
        print 'LED 3 off'
        GPIO.output(LEDPin3, GPIO.LOW)
        
                        
def destroy():
                GPIO.output(LEDPin1, GPIO.HIGH)
                GPIO.output(LEDPin2, GPIO.HIGH)
                GPIO.output(LEDPin3, GPIO.HIGH)
                GPIO.cleanup()

if __name__ == '__main__':
    setup()

    try:
                            while(True):
                                loop()
    except KeyboardInterrupt:
                                 destroy()

File: Lab1/Lab1_p2.c
#include <stdio.h>    // Used for printf() statements
#include <wiringPi.h> // Include WiringPi library!

// Pin number declarations. We're using the Broadcom chip pin numbers.
const int ledPin1 = 23; // Regular LED - Broadcom pin 23, P1 pin 16
const int ledPin2 = 24;
const int ledPin3 = 25;
void cleanup (void) __attribute__((destructor));

void cleanup (void)
{
    digitalWrite(ledPin1, 0);
    digitalWrite(ledPin2, 0);
    digitalWrite(ledPin3, 0);
    system("gpio unexport 23");
    system("gpio unexport 24");
    system("gpio unexport 25");
    

}


int main(void)
{
    // Setup stuff:
    wiringPiSetupGpio(); // Initialize wiringPi -- using Broadcom pin numbers
    pinMode(ledPin1, OUTPUT);     // Set regular LED as output
    pinMode(ledPin2, OUTPUT); 
    pinMode(ledPin3, OUTPUT); 

    digitalWrite(ledPin1, 0);//initialize pins low
    digitalWrite(ledPin2, 0);
    digitalWrite(ledPin3, 0);
	int i;
	for(i = 0; i < 5 ; i++){
		// write pin low
		digitalWrite(ledPin1, 1);
		delay(1000);
		digitalWrite(ledPin1, 0);
		digitalWrite(ledPin2, 1);
		delay(1000);
		digitalWrite(ledPin2, 0);
		digitalWrite(ledPin3, 1);
		delay(1000);
		digitalWrite(ledPin3, 0);
	}
    //digitalWrite(ledPin, LOW);     // Regular LED off
    //digitalWrite(ledPin, HIGH);     // Regular LED off
    digitalWrite(ledPin1, 0);
    digitalWrite(ledPin2, 0);
    digitalWrite(ledPin3, 0);
    system("gpio unexport 23");
    system("gpio unexport 24");
    system("gpio unexport 25");
    return 0;
}



